tinnsleep/create_reports.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+import mne
+from tinnsleep.config import Config
+from tinnsleep.data import CreateRaw, RawToEpochs_sliding
+from tinnsleep.classification import AmplitudeThresholding
+from tinnsleep.check_impedance import create_annotation_mne, Impedance_thresholding_sliding, check_RMS, \
+    fuse_with_classif_result
+from tinnsleep.signal import rms
+from tinnsleep.scoring import generate_clinical_report
+from tinnsleep.signal import is_good_epochs
+
+
+def preprocess(raw, picks_chan, picks_imp, duration, interval, params, THR_imp=6000, get_log=False):
+    """Preprocesses raw for reporting
+    Parameters
+    ----------
+    raw: Instance of mne.Raw
+        the signal to analyze
+    picks_chan: array-like, shape (n_channels,)
+        list of str corresponding of the name of each channel to analyze
+    picks_imp:
+        list of str corresponding of the name of each channel of impedance values associated with the channels to
+        analyze
+    duration : int
+        Number of elements (i.e. samples) on the epoch.
+    interval: int
+        Number of elements (i.e. samples) to move for the next epoch.
+    THR_imp: float
+        Threshold value for the impedance rejection algorithm
+    get_log : OPTIONAL boolean default False
+        if True, create a report of the preprocessing rejecting steps
+    Returns
+    -------
+    epochs : ndarray, shape (n_epochs, n_channels, duration)
+        Epoched view of `data`. Epochs are in the first dimension.
+    valid_labels : lsit of booleans
+        labels of the epochs as good (True) or bad (False) for future annotation and reporting
+    log : dictionary
+        logs of the preprocessing steps, including the number of epochs rejected at each step
+
+    """
+    # Epoch rejection based on impedance
+    check_imp = Impedance_thresholding_sliding(raw[picks_imp][0], duration, interval, THR=THR_imp)
+    impedance_labels = np.any(check_imp, axis=-1)
+    suppressed_imp = np.sum(impedance_labels)
+
+    raw = CreateRaw(raw[picks_chan][0], picks_chan, ch_types=['emg'])  # pick channels and load
+    # Filtering data
+    raw = raw.filter(20., 99., n_jobs=4,
+                     fir_design='firwin', filter_length='auto', phase='zero-double',
+                     picks=picks_chan)
+
+    # Creating epochs
+    epochs = RawToEpochs_sliding(raw, duration=duration, interval=interval)
+
+    # Epoch rejection based on |min-max| thresholding
+    amplitude_labels, bad_lists = is_good_epochs(epochs, **params)
+
+    suppressed_amp = np.sum(np.invert(amplitude_labels))
+
+    # Reuniting the rejection algorithms
+    valid_labels = np.invert(np.any(np.c_[impedance_labels, np.invert(amplitude_labels)], axis=-1))
```

Comment:
Promise this one works (less operations):
`valid_labels = np.all(np.c_[np.invert(impedance_labels), amplitude_labels], axis=-1))`

Reply:
Done!